StripeFinancialConnections/StripeFinancialConnections/Source/Native/NativeFlowController.swift

```diff
@@ -226,42 +225,52 @@ extension NativeFlowController {
         let completeFinancialConnectionsSession = { [weak self] in
             guard let self = self else { return }
             self.dataManager
-                .completeFinancialConnectionsSession()
+                .completeFinancialConnectionsSession(
+                    terminalError: customManualEntry ? "user_initiated_with_custom_manual_entry" : nil
+                )
                 .observe(on: .main) { [weak self] result in
                     guard let self = self else { return }
                     switch result {
                     case .success(let session):
                         let eventType = "object"
-                        if !session.accounts.data.isEmpty || session.paymentAccount != nil
-                            || session.bankAccountToken != nil
-                        {
+                        if session.status == .cancelled && session.statusDetails?.cancelled?.reason == .customManualEntry {
                             self.logCompleteEvent(
                                 type: eventType,
-                                status: "completed",
-                                numberOfLinkedAccounts: session.accounts.data.count
+                                status: "custom_manual_entry"
                             )
-                            finishAuthSession(.completed(session: session))
-                        } else if let closeAuthFlowError = closeAuthFlowError {
-                            self.logCompleteEvent(
-                                type: eventType,
-                                status: "failed",
-                                error: closeAuthFlowError
-                            )
-                            finishAuthSession(.failed(error: closeAuthFlowError))
+                            finishAuthSession(.failed(error: FinancialConnectionsSheetError.unknown(debugDescription: "custom manual entry")))
```

Comment:
So in this case users have to check the text to know it's manual entry?

Reply:
DM'd on this; this was temporary hack (PR in draft)